---
id: data-types
title: Data Types Overview
sidebar_label: Data Types and File Formats
---

ICGC ARGO provides access to data from ARGO member programs generated through the standardized [ARGO Analysis pipelines](/docs/analysis_workflows/data-types). The following page lists analyses and data files generated by the ARGO Analysis pipelines.

## Reads

### Submitted Reads

Raw reads submitted by ARGO member programs are used as inputs within the ARGO analytic pipeline. Data is accepted in FASTQ or BAM (aligned or unaligned) formats, must be compressed, and can be aligned or unaligned.

#### File Types

| Filename Pattern        | Description                             | Analysis Type    | Data Category    | Generating Workflow(s) |
| ----------------------- | --------------------------------------- | ---------------- | ---------------- | ---------------------- |
| \*.fq.gz, \*.fastq.gz   | FASTQ file format compressed with gzip  | sequencing_reads | Sequencing Reads | N/A                    |
| \*.fq.bz2, \*.fastq.bz2 | FASTQ file format compressed with bzip2 | sequencing_reads | Sequencing Reads | N/A                    |
| \*.bam                  | Binary Alignment Map (BAM) file format  | sequencing_reads | Sequencing Reads | N/A                    |

### Aligned Reads

A read is a sequence obtained from a single sequencing experiment. An aligned read is a sequence that has been aligned to a common reference genome. Typically these reads can number from the hundreds of thousands to tens of millions. Aligned reads are generated internally by the ARGO alignment workflow(s) and maintained in CRAM (a compressed version of BAM that stores only reads different from the reference sequence) format. ARGO reads are aligned to the GRCh38 Human Reference Genome.

#### File Types

| Filename Pattern | Description                               | Analysis Type    | Data Category    | Generating Workflow(s) |
| ---------------- | ----------------------------------------- | ---------------- | ---------------- | ---------------------- |
| \*.aln.cram      | CRAM file format and compressed with gzip | sequencing_reads | Sequencing Reads | DNA Seq Alignment      |

### Aligned Reads Index

These are secondary files that are external index files for CRAM format files. CRAI files follow a naming convention that corresponds to their matching CRAM file of the original filename suffixed with `.cram.crai`. Index files are required for selective access to genomic data inside a CRAM via CRAM slicing.

#### File Types

| Filename Pattern | Description                                                 | Analysis Type    | Data Category    | Generating Workflow(s) |
| ---------------- | ----------------------------------------------------------- | ---------------- | ---------------- | ---------------------- |
| \*.aln.cram.crai | CRAM Index file format. Requires a corresponding CRAM file. | sequencing_reads | Sequencing Reads | DNA Seq Alignment      |

## Quality Control Metrics

Quality control metrics are collected and recorded at several checkpoints in the ARGO Analysis pipeline to ensure that only high-quality data is released. All QC Metrics are associated to a corresponding analysis/file set that they annotate.

### Read Group QC

File containing read group (lane) level QC metrics.

#### File Types

| Filename Pattern       | Description                                    | Analysis Type | Data Category           | Generating Workflow(s) |
| ---------------------- | ---------------------------------------------- | ------------- | ----------------------- | ---------------------- |
| \*.ubam_qc_metrics.tgz | Generated by Picard CollectQualityYieldMetrics | qc_metrics    | Quality Control Metrics | DNA Seq Alignment      |

### Alignment QC

Files containing quality control metrics for aligned CRAM files.

#### File Types

| Filename Pattern   | Description                             | Analysis Type | Data Category           | Generating Workflow(s)                                 |
| ------------------ | --------------------------------------- | ------------- | ----------------------- | ------------------------------------------------------ |
| \*.qc_metrics.tgz  | Generated by Samtools stats             | qc_metrics    | Quality Control Metrics | DNA Seq Alignment                                      |
| \*.bas_metrics.tgz | Generated by Sanger bam_stats.pl script | qc_metrics    | Quality Control Metrics | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

### Duplicates Metrics

This is the data file containing duplicates metrics for aligned CRAM files. Multiple reads that match at the same position in the genome are located and tagged as duplicate reads in the CRAM file, where duplicate reads are defined as originating from a single fragment of DNA.

#### File Types

| Filename Pattern          | Description                                | Analysis Type | Data Category           | Generating Workflow(s) |
| ------------------------- | ------------------------------------------ | ------------- | ----------------------- | ---------------------- |
| \*.duplicates_metrics.tgz | Generated by biobambam2 bammarkduplicates2 | qc_metrics    | Quality Control Metrics | DNA Seq Alignment      |

### OxoG Metrics

Data file containing OxoG metrics, metrics quantifying the error rate resulting from oxidative artifacts, for aligned CRAM files.

#### File Types

| Filename Pattern    | Description                          | Analysis Type | Data Category           | Generating Workflow(s) |
| ------------------- | ------------------------------------ | ------------- | ----------------------- | ---------------------- |
| \*.oxog_metrics.tgz | Generated by GATK CollectOxoGMetrics | qc_metrics    | Quality Control Metrics | DNA Seq Alignment      |

### Ploidy and Purity Estimation

Data file containing purity and ploidy estimate.

#### File Types

| Filename Pattern     | Description                          | Analysis Type | Data Category           | Generating Workflow(s)     |
| -------------------- | ------------------------------------ | ------------- | ----------------------- | -------------------------- |
| \*.ascat_metrics.tgz | Estimated by Sanger ASCAT CNV caller | qc_metrics    | Quality Control Metrics | Sanger WGS Variant Calling |

### Cross Sample Contamination

Data file containing cross sample contamination estimate.

#### File Types

| Filename Pattern             | Description                                                                                                                                     | Analysis Type | Data Category           | Generating Workflow(s)     |
| ---------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------- | ----------------------- | -------------------------- |
| \*.contamination_metrics.tgz | Estimated by Sanger `verifyBamHomChk.pl` script, which provides information to determine whether the sample is possibly contaminated or swapped | qc_metrics    | Quality Control Metrics | Sanger WGS Variant Calling |

### Genotyped Gender Metrics

Data file containing genotypes comparison results of CRAM files from the same donor, including the fraction of matched genotypes and inferred donor gender.

#### File Types

| Filename Pattern                | Description                                          | Analysis Type | Data Category           | Generating Workflow(s)     |
| ------------------------------- | ---------------------------------------------------- | ------------- | ----------------------- | -------------------------- |
| \*.genotyped_gender_metrics.tgz | Generated by Sanger `compareBamGenotypes.pl` script. | qc_metrics    | Quality Control Metrics | Sanger WGS Variant Calling |

### Variant Calling Supplements

Data file containing variation results, temporary files, or runtime metric information generated by variant callers used in the ARGO analytic pipeline.

#### File Types

| Filename Pattern          | Description                                                                        | Analysis Type              | Data Category               | Generating Workflow(s)                                 |
| ------------------------- | ---------------------------------------------------------------------------------- | -------------------------- | --------------------------- | ------------------------------------------------------ |
| \*.ascat-supplement.tgz   | Generated by Sanger ASCAT CNV caller with bw files removed                         | variant_calling_supplement | Copy Number Variation       | Sanger WGS Variant Calling                             |
| \*.brass-supplement.tgz   | Generated by Sanger BRASS SV caller with bw files and intermediates.tar.gz removed | variant_calling_supplement | Structural Variation        | Sanger WGS Variant Calling                             |
| \*.caveman-supplement.tgz | Generated by Sanger CaVEMan SNV caller                                             | variant_calling_supplement | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |
| \*.pindel-supplement.tgz  | Generated by Sanger Pindel InDel caller                                            | variant_calling_supplement | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |
| \*.timings-supplement.tgz | Collected for different processing steps during the Sanger variant caller          | variant_calling_supplement | N/A                         | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

## Variant Calls

### Raw SNV Calls

Data files containing Single Nucleotide Variations (SNV) called from aligned reads, which have not yet been annotated but do have some filtering flags added.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category               | Generating Workflow(s)                                 |
| ---------------- | --------------- | --------------- | --------------------------- | ------------------------------------------------------ |
| \*.snv.vcf.gz    | VCF File format | variant_calling | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

### Raw SV Calls

Data files containing the structural variation data called from aligned reads, which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category        | Generating Workflow(s)     |
| ---------------- | --------------- | --------------- | -------------------- | -------------------------- |
| \*.sv.vcf.gz     | VCF File format | variant_calling | Structural Variation | Sanger WGS Variant Calling |

### Raw CNV Calls

Data files containing the copy number segments data called from aligned reads, which which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category         | Generating Workflow(s)     |
| ---------------- | --------------- | --------------- | --------------------- | -------------------------- |
| \*.cnv.vcf.gz    | VCF file format | variant_calling | Copy Number Variation | Sanger WGS Variant Calling |

### Raw InDel Calls

Data files containing the simple insertions and deletions (indels) data called from aligned reads, which have not yet been annotated but do have some filtering flags added.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category               | Generating Workflow(s)                                 |
| ---------------- | --------------- | --------------- | --------------------------- | ------------------------------------------------------ |
| \*.indel.vcf.gz  | VCF file format | variant_calling | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

### VCF Index

Secondary files that are external index files for VCF format files. TBI files follow a naming convention that corresponds to their matching VCF file of the original filename suffixed with `.vcf.gz.tbi`. Index files are required for selective access to genomic data inside a VCF file.
