---
id: dna-sanger-wgs-vc
title: Sanger WGS Variant Calling
sidebar_label: Sanger WGS Variant Calling
platform_key: DOCS_DNA_PIPELINE
---

Whole genome sequencing (WGS) aligned CRAM files are processed through the Sanger WGS Variant Calling Workflow as tumour/normal pairs. The ARGO DNA Seq pipeline has adopted the [Sanger Whole Genome Sequencing Analysis Docker Image](https://quay.io/wtsicgp/dockstore-cgpwgs:2.1.0) as the base workflow. For details, please see the latest version of the [ARGO Sanger WGS Variant Calling workflow](https://github.com/icgc-argo-workflows/sanger-wgs-variant-calling/releases).

## Inputs

- Normal WGS aligned CRAM and index files
- Tumour WGS aligned CRAM and index files
- [Reference files](ftp://ftp.sanger.ac.uk/pub/cancer/dockstore/human/GRCh38_hla_decoy_ebv)

## Processing

- `Pindel` InDel caller is used for somatic insertion/deletion variant detection.
- `ASCAT` CNV caller is used for somatic copy number variant analysis.
- `CaVEMan` SNV caller is used for somatic single nucleotide variant analysis.
- `BRASS` SV caller is used for somatic structural variation detection.

## Collect QC Metrics

- WGS aligned reads statistics are generated by [Sanger:bam_stats](https://github.com/ICGC-TCGA-PanCancer/PCAP-core/blob/master/bin/bam_stats.pl) script. The files containing normal/tumour aligned reads statistics are further used by Pindel and BRASS callers.
- Cross sample contamination is estimated by [Sanger:verifyBamHomChk](https://github.com/cancerit/cgpNgsQc/blob/develop/bin/verifyBamHomChk.pl) script for both normal and tumour samples.
- Purity and ploidy are estimated by `ASCAT` CNV caller
- Genotypes of CRAM files from the matched normal/tumour pair are compared and the fraction of matched genotypes are produced by [Sanger:compareBamGenotypes](https://github.com/cancerit/cgpNgsQc/blob/develop/bin/compareBamGenotypes.pl) script. It also checks if the inferred genders are matched.

## Outputs

- [Raw SNV Calls](/docs/data/variant-calls#raw-snv-calls) and [VCF Index](/docs/data/variant-calls#vcf-index)
- [Raw InDel Calls](/docs/data/variant-calls#raw-indel-calls) and [VCF Index](/docs/data/variant-calls#vcf-index)
- [Raw CNV Calls](/docs/data/variant-calls#raw-cnv-calls) and [VCF Index](/docs/data/variant-calls#vcf-index)
- [Raw SV Calls](/docs/data/variant-calls#raw-sv-calls) and [VCF Index](/docs/data/variant-calls#vcf-index)
- [SNV Supplement](/docs/data/variant-calls#snv-supplement) files
- [SV Supplement](/docs/data/variant-calls#sv-supplement) files
- [CNV Supplement](/docs/data/variant-calls#cnv-supplement) files
- [InDel Supplement](/docs/data/variant-calls#indel-supplement) files
- QC metrics files
  - [Alignment Metrics](/docs/data/qc-metrics#aligned-reads-qc) for both the Tumour and Normal samples
  - [Ploidy and Tumour Purity](/docs/data/qc-metrics#analysis-qc)
  - [Genotyping Stats](/docs/data/qc-metrics#analysis-qc)
  - [Cross Sample Contamination](/docs/data/qc-metrics#sample-qc)
  - [Runtime Stats](/docs/data/qc-metrics#analysis-qc)

## Workflow Diagram

![Sanger WGS Variant Calling Workflow](/assets/analysis-workflows/ARGO-WGS-variant-calling.png)
