---
id: dna-pipeline
title: DNA-Seq Analysis Pipeline
platform_key: DOCS_DNA_PIPELINE
---

The DNA-Sequencing (DNA-Seq) analysis pipeline identifies multiple types of somatic variant from both Whole Exome Sequencing (WXS) and Whole Genome Sequencing (WGS) sample data. DNA-Seq analysis is implemented across two main procedures:

- Sequence Alignment
- Variant Calling

In the future, these procedures will be extended to include:

- Variant Masking
- Variant Annotation
- Consensus Calling

## Alignment

The ARGO Data Platform accepts raw sequencing data in both FASTQ and BAM (aligned or unaligned) format. The first processing step in the DNA-Seq Pipeline is uniformly aligning samples to the GRCh38 reference genome. For details, please see the latest version of the [ARGO DNA Alignment](https://github.com/icgc-argo/dna-seq-processing-wfs/releases).

### Inputs

- All alignments are performed using [GRCh38](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome) as the human reference genome
- Submitted FASTQ or BAM files(s)

### Preprocessing

- Submitted sequencing reads (FASTQ or BAM) are converted into lane level (i.e read group level) BAMs.
- [Picard CollectQualityYieldMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360042475912-CollectQualityYieldMetrics-Picard-) is used for read group level BAM QC.

### Processing

- [BWA-MEM (version 0.7.17-r1188)](https://github.com/lh3/bwa/archive/v0.7.17.tar.gz) is performed to map the reads to the reference genome for each read group.
- [Biobambam (version 2.0.153)](https://gitlab.com/german.tischler/biobambam2/-/archive/2.0.153-release-20200124123734/biobambam2-2.0.153-release-20200124123734.tar.gz) is used to merge all read group level mapped BAMs into sample level BAM and mark duplicates per library.
- [Samtools stats](http://www.htslib.org/doc/samtools-stats.html) is used to calculate Alignment QC metrics.
- [Picard CollectOxoGMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360040098852-CollectOxoGMetrics-Picard-) is used to calculate the `OxoQ` score for oxidative artifact assessment.

### Outputs

- [Aligned read CRAM](/docs/data/reads#aligned-reads) and [index files](/docs/data/reads#aligned-reads-index)
- [Alignment QC metrics](/docs/data/qc-metrics#alignment-qc) files
- [OxoG Metrics](/docs/data/qc-metrics#oxog-metrics) files
- [Duplicates Metrics](/docs/data/qc-metrics#duplicates-metrics) files
- [Read Group QC Metrics](/docs/data/qc-metrics#read-group-qc) files for each submitted read group

![Alignment Workflow](/assets/analysis-workflows/ARGO-Alignment.png)

## Sanger WGS Variant Calling

Whole genome sequencing (WGS) aligned CRAM files are processed through the Sanger WGS Variant Calling Workflow as tumour/normal pairs. The ARGO DNA Seq pipeline has adopted the [Sanger Whole Genome Sequencing Analysis Docker Image](https://quay.io/wtsicgp/dockstore-cgpwgs:2.1.0) as the base workflow. For details, please see the latest version of the [ARGO Sanger WGS Variant Calling workflow](https://github.com/icgc-argo/sanger-wgs-variant-calling/releases).

### Inputs

- Normal WGS aligned CRAM and index files
- Tumour WGS aligned CRAM and index files
- [Reference files](ftp://ftp.sanger.ac.uk/pub/cancer/dockstore/human/GRCh38_hla_decoy_ebv)

### Processing

- `Pindel InDel caller` is used for somatic insertion/deletion variant detection.
- `ASCAT CNV caller` is used for somatic copy number variant analysis.
- `CaVEMan SNV caller` is used for somatic single nucleotide variant analysis.
- `BRASS SV caller` is used for somatic structural variation detection.

### Collect QC Metrics

- WGS aligned reads statistics are generated by Sanger [bam_stats.pl](https://github.com/ICGC-TCGA-PanCancer/PCAP-core/blob/master/bin/bam_stats.pl) script. The files containing normal/tumour aligned reads statistics are further used by Pindel and BRASS callers.
- Cross sample contamination is estimated by Sanger [verifyBamHomChk.pl](https://github.com/cancerit/cgpNgsQc/blob/develop/bin/verifyBamHomChk.pl) script for both normal and tumour samples.
- Purity and ploidy are estimated by `ASCAT CNV caller`
- Genotypes of CRAM files from the matched normal/tumour pair are compared and the fraction of matched genotypes are produced by Sanger [compareBamGenotypes.pl](https://github.com/cancerit/cgpNgsQc/blob/develop/bin/compareBamGenotypes.pl) script. It also checks if the inferred genders are matched.

### Outputs

- [SNV VCF](/docs/data/variant-calls#raw-snv-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [InDel VCF](/docs/data/variant-calls#raw-indel-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [CNV VCF](/docs/data/variant-calls#raw-cnv-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [SV VCF](/docs/data/variant-calls#raw-sv-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [Variant calling supplement](/docs/data/variant-calls#variant-calling-supplement) files
- QC metrics files
  - [Alignment QC](/docs/data/qc-metrics#alignment-qc) for both the Tumour and Normal samples
  - [Ploidy and Purity Estimation](/docs/data/qc-metrics#ploidy-and-purity-estimation)
  - [Genotyping Inferred Gender](/docs/data/qc-metrics#genotyping-inferred-gender)
  - [Cross Sample Contamination](/docs/data/qc-metrics#cross-sample-contamination)

![Sanger WGS Variant Calling Workflow](/assets/analysis-workflows/ARGO-WGS-variant-calling.png)

## Sanger WXS Variant Calling

Whole exome sequencing (WXS) aligned CRAM files are processed through the Sanger WXS Variant Calling Workflow as tumour/normal pairs. The ARGO DNA Seq pipeline has adopted the [Sanger Whole Exome Sequencing Analysis Docker Image](https://quay.io/wtsicgp/dockstore-cgpwxs:3.1.6) as the base workflow. For details, please see the latest version of the [ARGO Sanger WXS Variant Calling workflow](https://github.com/icgc-argo/sanger-wxs-variant-calling/releases).

### Inputs

- Normal WXS aligned CRAM and index files
- Tumour WXS aligned CRAM and index files
- [Reference files](ftp://ftp.sanger.ac.uk/pub/cancer/dockstore/human/GRCh38_hla_decoy_ebv)

### Processing

- `Pindel InDel caller` is used for somatic insertion/deletion variant detection.
- `CaVEMan SNV caller` is used for somatic single nucleotide variant analysis.

### Collect QC Metrics

- WXS aligned reads statistics are generated by Sanger [bam_stats.pl](https://github.com/ICGC-TCGA-PanCancer/PCAP-core/blob/master/bin/bam_stats.pl) script. The files containing normal/tumour aligned reads statistics are further used by Pindel caller.

### Outputs

- [SNV VCF](/docs/data/variant-calls#raw-snv-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [InDel VCF](/docs/data/variant-calls#raw-indel-calls) and [index files](/docs/data/variant-calls#vcf-index)
- [Variant calling supplement](/docs/data/variant-calls#variant-calling-supplements) files
- QC Metrics
  - [Alignment QC](/docs/data/reads#alignment-qc) for both the Tumour and Normal samples

![Sanger WXS Variant Calling Workflow](/assets/analysis-workflows/ARGO-WXS-variant-calling.png)
