---
id: variant-calls
title: Variant Calls
sidebar_label: Variant Calls
---

ICGC ARGO provides access to data from ARGO member programs generated through the standardized [ARGO Analysis pipelines](/docs/analysis-workflows/analysis-overview). The following page lists variant calling analyses and data files generated by the ARGO Analysis pipelines.

## Raw SNV Calls

Data files containing Single Nucleotide Variations (SNV) called from aligned reads, which have not yet been annotated but do have some filtering flags added.

#### File Types

| Filename Pattern | Description                     | Analysis Type   | Data Category               | Generating Workflow(s)                                                                                                  |
| ---------------- | ------------------------------- | --------------- | --------------------------- | ----------------------------------------------------------------------------------------------------------------------- |
| \*.snv.vcf.gz    | SNV results in VCF File format. | variant_calling | Simple Nucleotide Variation | <ul><li>Sanger WGS Variant Calling </li><li>Sanger WXS Variant Calling </li><li> GATK Mutect2 Variant Calling</li></ul> |

## SNV Supplement

Data files containing SNV results and temporary files generated by variant callers used in the ARGO Analysis pipeline.

#### File Types

| Filename Pattern          | Description                                        | Analysis Type              | Data Category               | Generating Workflow(s)                                                            |
| ------------------------- | -------------------------------------------------- | -------------------------- | --------------------------- | --------------------------------------------------------------------------------- |
| \*.caveman-supplement.tgz | Data files generated by Sanger CaVEMan SNV caller. | variant_calling_supplement | Simple Nucleotide Variation | <ul><li>Sanger WGS Variant Calling </li><li>Sanger WXS Variant Calling </li></ul> |

## Raw SV Calls

Data files containing Structural Variations (SV) called from aligned reads, which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description                    | Analysis Type   | Data Category        | Generating Workflow(s)     |
| ---------------- | ------------------------------ | --------------- | -------------------- | -------------------------- |
| \*.sv.vcf.gz     | SV results in VCF File format. | variant_calling | Structural Variation | Sanger WGS Variant Calling |

## SV Supplement

Data files containing SV results and temporary files generated by variant callers used in the ARGO Analysis pipeline.

#### File Types

| Filename Pattern        | Description                                                                                    | Analysis Type              | Data Category        | Generating Workflow(s)     |
| ----------------------- | ---------------------------------------------------------------------------------------------- | -------------------------- | -------------------- | -------------------------- |
| \*.brass-supplement.tgz | Data files generated by Sanger BRASS SV caller with bw files and intermediates.tar.gz removed. | variant_calling_supplement | Structural Variation | Sanger WGS Variant Calling |

## Raw CNV Calls

Data files containing the Copy Number Varations (CNV) called from aligned reads, which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description                     | Analysis Type   | Data Category         | Generating Workflow(s)     |
| ---------------- | ------------------------------- | --------------- | --------------------- | -------------------------- |
| \*.cnv.vcf.gz    | CNV results in VCF file format. | variant_calling | Copy Number Variation | Sanger WGS Variant Calling |

## CNV Supplement

Data files containing CNV results and temporary files generated by variant callers used in the ARGO Analysis pipeline.

#### File Types

| Filename Pattern        | Description                                                            | Analysis Type              | Data Category         | Generating Workflow(s)     |
| ----------------------- | ---------------------------------------------------------------------- | -------------------------- | --------------------- | -------------------------- |
| \*.ascat-supplement.tgz | Data files generated by Sanger ASCAT CNV caller with bw files removed. | variant_calling_supplement | Copy Number Variation | Sanger WGS Variant Calling |

## Raw InDel Calls

Data files containing the simple Insertions and Deletions (InDel) data called from aligned reads, which have not yet been annotated but do have some filtering flags added.

## InDel Supplement

Data files containing InDel results and temporary files generated by variant callers used in the ARGO Analysis pipeline.

#### File Types

| Filename Pattern         | Description                                         | Analysis Type              | Data Category               | Generating Workflow(s)                                                            |
| ------------------------ | --------------------------------------------------- | -------------------------- | --------------------------- | --------------------------------------------------------------------------------- |
| \*.pindel-supplement.tgz | Data files generated by Sanger Pindel InDel caller. | variant_calling_supplement | Simple Nucleotide Variation | <ul><li>Sanger WGS Variant Calling </li><li>Sanger WXS Variant Calling </li></ul> |

#### File Types

| Filename Pattern | Description                       | Analysis Type   | Data Category               | Generating Workflow(s)                                                                                                  |
| ---------------- | --------------------------------- | --------------- | --------------------------- | ----------------------------------------------------------------------------------------------------------------------- |
| \*.indel.vcf.gz  | InDel results in VCF file format. | variant_calling | Simple Nucleotide Variation | <ul><li>Sanger WGS Variant Calling </li><li>Sanger WXS Variant Calling </li><li> GATK Mutect2 Variant Calling</li></ul> |

## Filtered SNV Calls

Data files containing Single Nucleotide Variations (SNV) which were filtered through each caller's PASS tag and located only within the open access regions.

#### File Types

| Filename Pattern          | Description                                          | Analysis Type      | Data Category               | Generating Workflow(s)                 |
| ------------------------- | ---------------------------------------------------- | ------------------ | --------------------------- | -------------------------------------- |
| \*.snv.open-filter.vcf.gz | Open access filtered SNV results in VCF File format. | variant_processing | Simple Nucleotide Variation | <ul>Open Access Variant Filtering</ul> |

## Filtered InDel Calls

Data files containing the simple Insertions and Deletions (InDel) variants which were filtered through each caller's PASS tag and located only within the open access regions.

#### File Types

| Filename Pattern            | Description                                            | Analysis Type      | Data Category               | Generating Workflow(s)                 |
| --------------------------- | ------------------------------------------------------ | ------------------ | --------------------------- | -------------------------------------- |
| \*.indel.open-filter.vcf.gz | Open access filtered InDel results in VCF file format. | variant_processing | Simple Nucleotide Variation | <ul>Open Access Variant Filtering</ul> |

## VCF Index

Secondary files that are external index files for VCF format files. TBI files follow a naming convention that corresponds to their matching VCF file of the original filename suffixed with `.vcf.gz.tbi`. Index files are required for selective access to genomic data inside a VCF file.

#### File Types

| Filename Pattern | Description                                               | Analysis Type                                                | Data Category                                                                                            | Generating Workflow(s)                                                                                                                                        |
| ---------------- | --------------------------------------------------------- | ------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| \*.vcf.gz.tbi    | VCF Index file format. Requires a corresponding VCF file. | <ul><li>variant_calling</li><li>variant_processing</li></ul> | <ul><li>Simple Nucleotide Variation</li><li>Structural Variation</li><li>Copy Number Variation</li></ul> | <ul><li>Sanger WGS Variant Calling </li><li>Sanger WXS Variant Calling </li><li> GATK Mutect2 Variant Calling</li><li>Open Access Variant Filtering</li></ul> |
