---
id: variant-calls
title: Variant Calls
sidebar_label: Variant Calls
---

ICGC ARGO provides access to data from ARGO member programs generated through the standardized [ARGO Analysis pipelines](/docs/analysis-workflows/analysis-overview). The following page lists variant calling analyses and data files generated by the ARGO Analysis pipelines.

## Raw SNV Calls

Data files containing Single Nucleotide Variations (SNV) called from aligned reads, which have not yet been annotated but do have some filtering flags added.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category               | Generating Workflow(s)                                 |
| ---------------- | --------------- | --------------- | --------------------------- | ------------------------------------------------------ |
| \*.snv.vcf.gz    | VCF File format | variant_calling | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

## Raw SV Calls

Data files containing the structural variation data called from aligned reads, which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category        | Generating Workflow(s)     |
| ---------------- | --------------- | --------------- | -------------------- | -------------------------- |
| \*.sv.vcf.gz     | VCF File format | variant_calling | Structural Variation | Sanger WGS Variant Calling |

## Raw CNV Calls

Data files containing the copy number segments data called from aligned reads, which have not yet been annotated or filtered.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category         | Generating Workflow(s)     |
| ---------------- | --------------- | --------------- | --------------------- | -------------------------- |
| \*.cnv.vcf.gz    | VCF file format | variant_calling | Copy Number Variation | Sanger WGS Variant Calling |

## Raw InDel Calls

Data files containing the simple insertions and deletions (indels) data called from aligned reads, which have not yet been annotated but do have some filtering flags added.

#### File Types

| Filename Pattern | Description     | Analysis Type   | Data Category               | Generating Workflow(s)                                 |
| ---------------- | --------------- | --------------- | --------------------------- | ------------------------------------------------------ |
| \*.indel.vcf.gz  | VCF file format | variant_calling | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |

## VCF Index

Secondary files that are external index files for VCF format files. TBI files follow a naming convention that corresponds to their matching VCF file of the original filename suffixed with `.vcf.gz.tbi`. Index files are required for selective access to genomic data inside a VCF file.

#### File Types

| Filename Pattern | Description                                               | Analysis Type   | Data Category                                                                                       | Generating Workflow(s) |
| ---------------- | --------------------------------------------------------- | --------------- | --------------------------------------------------------------------------------------------------- | ---------------------- |
| \*.vcf.gz.tbi    | VCF Index file format. Requires a corresponding VCF file. | variant_calling | Simple Nucleotide Variation, Structural Variation, or Copy Number Variation depending on parent VCF | DNA Seq Alignment      |

## Variant Calling Supplement

Data files containing variation results, temporary files, or runtime metric information generated by variant callers used in the ARGO Analysis pipeline.

#### File Types

| Filename Pattern          | Description                                                                        | Analysis Type              | Data Category               | Generating Workflow(s)                                 |
| ------------------------- | ---------------------------------------------------------------------------------- | -------------------------- | --------------------------- | ------------------------------------------------------ |
| \*.ascat-supplement.tgz   | Generated by Sanger ASCAT CNV caller with bw files removed                         | variant_calling_supplement | Copy Number Variation       | Sanger WGS Variant Calling                             |
| \*.brass-supplement.tgz   | Generated by Sanger BRASS SV caller with bw files and intermediates.tar.gz removed | variant_calling_supplement | Structural Variation        | Sanger WGS Variant Calling                             |
| \*.caveman-supplement.tgz | Generated by Sanger CaVEMan SNV caller                                             | variant_calling_supplement | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |
| \*.pindel-supplement.tgz  | Generated by Sanger Pindel InDel caller                                            | variant_calling_supplement | Simple Nucleotide Variation | Sanger WGS Variant Calling, Sanger WXS Variant Calling |
| \*.timings-supplement.tgz | Collected for different processing steps during the Sanger variant caller          | variant_calling_supplement | N/A                         | Sanger WGS Variant Calling, Sanger WXS Variant Calling |
